packages/babel-plugin-proposal-object-rest-spread/src/index.js

```diff
@@ -346,7 +346,11 @@ export default declare((api, opts) => {
           );
           refPropertyPath.forEach(prop => {
             const { node } = prop;
-            ref = t.memberExpression(ref, t.cloneNode(node.key), node.computed);
+            ref = t.memberExpression(
+              ref,
+              t.cloneNode(node.key),
+              t.isLiteral(node.key) || node.computed,
```

Comment:
nit:
```suggestion
              node.computed || t.isLiteral(node.key),
```
is more intuitive to me and slightly more efficient.